src/climate_learn/models/module.py

```diff
@@ -0,0 +1,103 @@
+# Standard library
+from typing import Any, Callable, Dict, List, Optional
+
+# Third party
+import torch
+from torch.optim.lr_scheduler import _LRScheduler as LRScheduler
+import pytorch_lightning as pl
+
+
+class LitModule(pl.LightningModule):
+    def __init__(
+        self,
+        net: torch.nn.Module,
+        optimizer: torch.optim.Optimizer,
+        lr_scheduler: LRScheduler,
+        train_loss: Callable,
+        val_loss: List[Callable],
+        test_loss: List[Callable],
+        train_target_transform: Optional[Callable] = None,
+        val_target_transforms: Optional[List[Callable]] = None,
+        test_target_transforms: Optional[List[Callable]] = None,
+    ):
+        super().__init__()
+        self.net = net
+        self.optimizer = optimizer
+        self.lr_scheduler = lr_scheduler
+        self.train_loss = train_loss
+        self.val_loss = val_loss
+        self.test_loss = test_loss
+        self.train_target_transform = train_target_transform
+        self.val_target_transform = val_target_transforms
+        self.test_target_transform = test_target_transforms
+
+    def forward(self, x):
+        return self.net(x)
+
+    def training_step(self, batch: Any, batch_idx: int):
+        x, y, in_variables, out_variables = batch
+        yhat = self(x).to(device=y.device)
+        if self.train_target_transform:
+            yhat = self.train_target_transform(yhat)
+            y = self.train_target_transform(y)
+        losses = self.train_loss(yhat, y)
+        loss_name = getattr(self.train_loss, "name", "loss")
+        if losses.dim() == 0:  # aggregate loss only
+            loss = losses
+        else:  # per channel + aggregate
+            loss_dict = {}
+            for var_name, loss in zip(out_variables, losses):
+                loss_dict[f"{loss_name}:{var_name}"] = loss
+            self.log_dict(
+                loss_dict,
+                prog_bar=True,
+                on_step=True,
+                on_epoch=False,
+                batch_size=x.shape[0],
+            )
+            loss = losses[-1]
+        return loss
+
+    def validation_step(self, batch: Any, batch_idx: int):
+        self.evaluate(batch, "val")
+
+    def test_step(self, batch: Any, batch_idx: int):
+        self.evaluate(batch, "test")
+
+    def evaluate(self, batch, stage):
+        x, y, in_variables, out_variables = batch
+        yhat = self(x).to(device=y.device)
+        loss_fns = self.val_loss
+        loss_dict = {}
+        for i, lf in enumerate(loss_fns):
+            if stage == "val" and self.val_target_transform is not None:
+                yhat_T = self.val_target_transform[i](yhat)
```

Comment:
You have correctly ascertained the correct behavior. I aim to provide as much flexibility to the user as possible so they can pick any combination of transforms and metrics. Besides, in the future, we should definitely change the data code so that the user can specify whatever transform they want since normalization is not appropriate for all modalities of climate data. I will update my code to make this intended purpose more clear.